### ISSUE TITLE
ec2 instance - chef provisioner - windows

### ISSUE THREAD

**Issue by swade1987** [hashicorp/terraform#5519]
I have the following terraform provisioner:

```
provisioner "chef"  {
        connection {
          user = "steven"
          password = "XYZ"
          type = "winrm"
          timeout = "30m"
        }
        environment = "system-test"
        run_list = ["Parliament.CDP.Web::default"]
        node_name = "cd-demo-web-01"
        server_url = "https://api.chef.io/organizations/parliamentary-digital-service"
        validation_client_name = "parliamentary-digital-service-validator"
        validation_key = "${file(\"parliamentary-digital-service-validator.pem\")}"
    }
```

However, terraform connects via WinRM and then throws the following error:

```
aws_instance.cd-demo-web-01 (chef): Connecting to remote host via WinRM...
aws_instance.cd-demo-web-01 (chef):   Host: 52.37.228.110
aws_instance.cd-demo-web-01 (chef):   Port: 5985
aws_instance.cd-demo-web-01 (chef):   User: steven
aws_instance.cd-demo-web-01 (chef):   Password: true
aws_instance.cd-demo-web-01 (chef):   HTTPS: false
aws_instance.cd-demo-web-01 (chef):   Insecure: false
aws_instance.cd-demo-web-01 (chef):   CACert: false
aws_instance.cd-demo-web-01 (chef): Connected!
Error applying plan:

1 error(s) occurred:

* Uploading client.rb failed: Error restoring file from $env:TEMP\winrmcp-56df3e9a-692b-b098-7964-afb82751ad8b.tmp to C:\Temp\ChefClient.ps1: restore operation returned code=1

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

I removed the version and now it works, however this is more than likely a one off event, its really not consistent.

**Comment by swade1987** [comment]
created another ec2 instance, now I am getting the same error

**Comment by svanharmelen** [comment]
@swade1987 hmm... This seems very annoying... Did you manage to find any way to reproduce this consistently?

**Comment by svanharmelen** [comment]
@swade1987 and? Did you manage to find a way to reproduce this or find a root cause?

**Comment by svanharmelen** [comment]
@swade1987 I'm tempted to close this issue as I cannot reproduce this. Any thoughts about that?

**Comment by swade1987** [comment]
@svanharmelen sounds good to me :)

**Comment by svanharmelen** [comment]
Cool, thanks...

**Comment by strata-alex** [comment]
I've got similar issue 
```
aws_instance.sqlserver: Still creating... (4m21s elapsed)
Error applying plan:

1 error(s) occurred:

* Uploading client.rb failed: Error removing temporary file $env:TEMP\winrmcp-57715e5b-ba34-abe1-9e34-09f20650860b.tmp: http error: 401 -

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

interesting here is $env:TMP variable should be `%USERPROFILE%\AppData\Local\Temp` according to System Properties - Environment Variables - User variables, but when I do ` cd $env:TEMP` in poweshell or cmd.exe I get to `C:\Users\Administrator\AppData\Local\Temp\2>` - notice "2" here and the file `winrmcp-57715e5b-ba34-abe1-9e34-09f20650860b.tmp` is located in `PS C:\Users\Administrator\AppData\Local\Temp\>` (without 2) folder.

BTW I'm using 0.6.16.

**Comment by netflash** [comment]
I've got similar issue 
```
aws_instance.sqlserver: Still creating... (4m21s elapsed)
Error applying plan:

1 error(s) occurred:

* Uploading client.rb failed: Error removing temporary file $env:TEMP\winrmcp-57715e5b-ba34-abe1-9e34-09f20650860b.tmp: http error: 401 -

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

interesting here is $env:TMP variable should be `%USERPROFILE%\AppData\Local\Temp` according to System Properties - Environment Variables - User variables, but when I do ` cd $env:TEMP` in poweshell or cmd.exe I get to `C:\Users\Administrator\AppData\Local\Temp\2>` - notice "2" here and the file `winrmcp-57715e5b-ba34-abe1-9e34-09f20650860b.tmp` is located in `PS C:\Users\Administrator\AppData\Local\Temp\>` (without 2) folder.

PS terraform version 0.6.16.
PPS I can reproduce this - it goes three times in a row with different newly-created instances.

**Comment by Vermillionaire** [comment]
I am also having an issue with this.

From my testing it seems to happen to 1 in 4 machines randomly when the chef provisoner is running. Here are a few examples I've encountered:

> Uploading first-boot.json failed: Error uploading file to $env:TEMP\winrmcp-57fb7bc9-1da3-a301-2af8-cfe942b22818.tmp: Couldn't create shell: http error: 401

> WARNING: Failed to cleanup user key on new node: Error executing command "cd C:/chef && del /F /Q validator.pem": http error: 401 -

> Uploading client.rb failed: Error restoring file from $env:TEMP\winrmcp-57fb3601-8307-aa1d-72a3-80316b314e27.tmp to C:\Temp\ChefClient.ps1: restore operation returned code=1

Last time I had the "Error restoring file" I took a look at the file uploaded to the machine in the temp folder, and it looks like the file uploaded is corrupted.  The other errors seem indicate that winrm failed completely to copy the file. Also worth noting is that this issue can happen with the remote-exe provisioner too. I tried to do a simple remote "dir" command and the cmd file terraform copied was also corrupted.

This is what I'm running on:
Terraform Version: 0.7.5
Windows 7 Enterprise
WinRM Verion: 3.0
Powershell Versoin 4.0

This was the only issue I could find related to my problem in the terraform issues (there seem to be a lot of similar issues in the packer github, but they haven't really helped). Does anyone have a solution or suggestions on how to resolve this?
